tool_test.go

```diff
@@ -0,0 +1,161 @@
+package codacytool
+
+import (
+	"fmt"
+	logrus "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"io/ioutil"
+
+	"os"
+	"strings"
+	"testing"
+)
+
+type ToolTestSuite struct {
+	suite.Suite
+}
+
+func TestExampleTestSuite(t *testing.T) {
+	suite.Run(t, new(ToolTestSuite))
+}
+
+func (suite *ToolTestSuite) SetupTest() {
+	toolConfigsBasePathFlag = &testsResourcesLocation
+}
+
+func testingTool(name, version string) (ToolDefinition, string) {
+	patternObj, patternJSON := pattern()
+	toolRepresentationAsJSON := fmt.Sprintf(`{"name":"%s","version":"%s","patterns":[%s]}`, name, version, patternJSON)
+	if version == "" {
+		toolRepresentationAsJSON = fmt.Sprintf(`{"name":"%s","patterns":[%s]}`, name, patternJSON)
+	}
+	return ToolDefinition{
+		Name:     name,
+		Version:  version,
+		Patterns: []Pattern{patternObj},
+	}, toolRepresentationAsJSON
+}
+
+func (suite *ToolTestSuite) TestToolToJSON() {
+	name := "govet"
+	version := "0.0.1"
+	tool, toolJSONExpected := testingTool(name, version)
+	toolAsJSON, err := tool.ToJSON()
+
+	assert.Nil(suite.T(), err, "Failed converting to JSON")
+	assert.Equal(suite.T(), toolJSONExpected, string(toolAsJSON), "Failed converting to JSON")
+}
+
+func (suite *ToolTestSuite) TestToolToJSONWithoutVersion() {
+	name := "govet"
+	tool, toolJSONExpected := testingTool(name, "")
+	toolAsJSON, err := tool.ToJSON()
+
+	assert.Nil(suite.T(), err, "Failed converting to JSON")
+	assert.Equal(suite.T(), toolJSONExpected, string(toolAsJSON), "Failed converting to JSON")
+}
+
+func (suite *ToolTestSuite) TestLoadToolDefinition() {
+	patternsFileLocation := defaultDefinitionFile()
+	tool, err := LoadToolDefinition(patternsFileLocation)
+
+	assert.Nil(suite.T(), err, "Failed to load tool %s", patternsFileLocation)
+	assert.Equal(suite.T(), "govet", tool.Name)
+
+	numPatterns := len(tool.Patterns)
+	expectedPatterns := 1
+
+	assert.Equal(suite.T(), expectedPatterns, numPatterns)
+}
+
+func (suite *ToolTestSuite) TestPrintResults() {
+	r, w, oldStdout := setStdoutToBuffer()
+
+	defer func() {
+		logrus.SetOutput(os.Stderr)
```

Comment:
what is this line doing?

Reply:
This line is not needed. Removed it
Nice catch 👌